test/emitter.test.ts

```diff
@@ -0,0 +1,88 @@
+// Note: we need to import logger from the root
+// because this is the logger used in logError in ../src/common/util
+import { logger } from "../node_modules/@coder/logger"
+
+import { Emitter } from "../src/common/emitter"
+
+describe("emitter", () => {
+  describe("Emitter", () => {
+    let spy: jest.SpyInstance
+
+    beforeEach(() => {
+      spy = jest.spyOn(logger, "error")
+    })
+
+    afterEach(() => {
+      jest.clearAllMocks()
+    })
+
+    afterAll(() => {
+      jest.restoreAllMocks()
+    })
+
+    it("should return an Emitter", async () => {
```

Comment:
I think "should run the correct callbacks" would be a more apt description for the test? Not sure :sweat_smile:

Reply:
good idea! let's change it to that!